lib/3scale/backend/stats/keys/generator/partition.rb

```diff
@@ -0,0 +1,47 @@
+# Create a keys partition generator
+module ThreeScale
+  module Backend
+    module Stats
+      module Keys
+        module Generator
+          class Partition
+            include Enumerable
+
+            private
+
+            attr_accessor :keys_slice
+            attr_accessor :key_generator
+            attr_accessor :key_partition_generator
+
+            public
+
+            # It is 20*50 because we want 20 batches of 50 keys
+            # to delete in Redis in the resque deletion job
+            DEFAULT_KEYS_SLICE = 20*50
+            private_constant :DEFAULT_KEYS_SLICE
+
+            def initialize(key_generator, options = {})
+              self.keys_slice = options[:keys_slice] || DEFAULT_KEYS_SLICE
+              self.key_generator = key_generator
+              self.key_partition_generator = Enumerator.new do |enum|
```

Comment:
you don't need this - this is an Enumerator on top of another one to adapt the output. If you don't want to execute it ASAP you need to call `lazy` on any `Enumerable`.